federatedscope/cl/fedgc/server.py

```diff
@@ -0,0 +1,252 @@
+import torch
+import logging
+import copy
+import numpy as np
+
+from federatedscope.core.message import Message
+from federatedscope.core.workers.server import Server
+from federatedscope.core.auxiliaries.utils import merge_dict
+from federatedscope.cl.fedgc.utils import global_NT_xentloss
+from torchviz import make_dot, make_dot_from_trace
+
+logger = logging.getLogger(__name__)
+
+
+class GlobalContrastFLServer(Server):
+    r"""
+    GlobalContrastFL(Fedgc) Server contain two part in training: Fedavg aggragator 
+    for client model weight and calculate global loss from all sampled client embedding
+    then broadcast all client to train model.
+    """
+    def __init__(self,
+                 ID=-1,
+                 state=0,
+                 config=None,
+                 data=None,
+                 model=None,
+                 client_num=5,
+                 total_round_num=10,
+                 device='cpu',
+                 strategy=None,
+                 **kwargs):
+        super(GlobalContrastFLServer,
+              self).__init__(ID, state, config, data, model, client_num,
+                             total_round_num, device, strategy, **kwargs)
+        # Initial seqs_embedding
+        self.seqs_embedding = {
+            idx: ()
+            for idx in range(1, self._cfg.federate.client_num + 1)
+        }
+        self.loss_list= {
+            idx: 0
+            for idx in range(1, self._cfg.federate.client_num + 1)
+        }
+    
+    def _register_default_handlers(self):
```

Comment:
is it possible to call base class's method to execute the first four registrations?